src/modules/post/entity.rs

```diff
@@ -1,15 +1,32 @@
-use async_graphql::SimpleObject;
+use async_graphql::{Enum, SimpleObject};
 use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
 use uuid::Uuid;
 
 use crate::modules::user::User;
 
-#[derive(Debug, Deserialize, Serialize, SimpleObject)]
+#[derive(Copy, Clone, Debug, Deserialize, Enum, PartialEq, Eq, Serialize, sqlx::Type)]
+#[sqlx(rename_all = "lowercase")]
+pub enum Scope {
+    Public,
+    Private,
+}
+
+impl ToString for Scope {
+    fn to_string(&self) -> String {
+        match &self {
+            Self::Public => String::from("public"),
+            Self::Private => String::from("private"),
+        }
+    }
+}
```

Comment:
Not sure if after adding the `#[sqlx]` macro on L9 the implementation of `ToString` is still required. Have you tried w/o it?